src/Parser.php

```diff
@@ -127,13 +127,16 @@ private function disjunction()
         $terms = [];
         $terms[] = $this->conjunction();
 
-        if ($this->lexer->isNextToken(Tokens::T_SP)) {
+        $isNextTokenOr = true;
+        while($this->lexer->isNextToken(Tokens::T_SP) && $isNextTokenOr) {
             $nextToken = $this->lexer->glimpse();
             if ($this->isName('or', $nextToken)) {
                 $this->match(Tokens::T_SP);
                 $this->match(Tokens::T_NAME);
                 $this->match(Tokens::T_SP);
                 $terms[] = $this->conjunction();
+            } else {
+                $isNextTokenOr = false;
             }
         }
```

Comment:
Instead of using a variable to break the `while` loop, simply use the `break` statement instead.

```php
while ($this->lexer->isNextToken(Tokens::T_SP)) {
    $nextToken = $this->lexer->glimpse();
    
    if ($this->isName('or', $nextToken) === false) {
        break;
    }
    
    $this->match(Tokens::T_SP);
    $this->match(Tokens::T_NAME);
    $this->match(Tokens::T_SP);
    $terms[] = $this->conjunction();
}
```